### PR TITLE
fix(components): [select] style break change in multiple collapseTags

### LINKED ISSUE #15459: [Component] [select] 升级2.5.0后，select组件设置 multiple collapse-tags时，当组件宽度不足时，样式发生了破坏性的变化。
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.5.0`
- Browser / OS: `UserAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-select`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IHZhbHVlID0gcmVmKFsnT3B0aW9uMScsICdPcHRpb24yJ10pXG5cbmNvbnN0IG9wdGlvbnMgPSBbXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjEnLFxuICAgIGxhYmVsOiAnT3B0aW9uMScsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjInLFxuICAgIGxhYmVsOiAnT3B0aW9uMicsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjMnLFxuICAgIGxhYmVsOiAnT3B0aW9uMycsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjQnLFxuICAgIGxhYmVsOiAnT3B0aW9uNCcsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjUnLFxuICAgIGxhYmVsOiAnT3B0aW9uNScsXG4gIH0sXG5dXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZWwtc2VsZWN0XG4gICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICBtdWx0aXBsZVxuICAgIGNvbGxhcHNlLXRhZ3NcbiAgICBwbGFjZWhvbGRlcj1cIlNlbGVjdFwiXG4gICAgc3R5bGU9XCJ3aWR0aDogMTQwcHhcIlxuICAgIGNvbGxhcHNlLXRhZ3MtdG9vbHRpcFxuICAgIDptYXgtY29sbGFwc2UtdGFncz1cIjFcIlxuICA+XG4gICAgPGVsLW9wdGlvblxuICAgICAgdi1mb3I9XCJpdGVtIGluIG9wdGlvbnNcIlxuICAgICAgOmtleT1cIml0ZW0udmFsdWVcIlxuICAgICAgOmxhYmVsPVwiaXRlbS5sYWJlbFwiXG4gICAgICA6dmFsdWU9XCJpdGVtLnZhbHVlXCJcbiAgICAvPlxuICA8L2VsLXNlbGVjdD5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

### Steps to reproduce

切换element-plus版本至2.5.0版本，发现组件内tags出现换行，切换回2.4.4版本时不会换行

### What is Expected?

组件内tags不换行

### What is actually happening?

组件内tags换行

### Additional comments

一般来讲，传collapse-tags且max-collapse-tags为1，都是希望不要出现换行导致的组件高度发现变化，这样会导致样式布局出现问题。所以这里还是保持和2.4.4版本一致为好。

image
image

### PR DESCRIPTION
**Pull request by Fuphoenixes**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fixed: https://github.com/element-plus/element-plus/issues/15459

### CODE REVIEW

**Comment by Fuphoenixes** [comment]
@kooriookami see see

**Review by kooriookami — APPROVED** [review]

**Comment by xiaoxian521** [comment]
`el-select-v2` also has the same problem

image

**Comment by kooriookami** [comment]
Yes, please add this feature to select-v2 too.

**Comment by Fuphoenixes** [comment]
done.

**Review by kooriookami — APPROVED** [review]

**Comment by kooriookami** [comment]
Nice~

**Comment by Fuphoenixes** [comment]
Wait, I found a problem.
image
what happened?

**Comment by Fuphoenixes** [comment]
There are similar questions in the docs. 
image

**Comment by Fuphoenixes** [comment]
ok

File: packages/components/select-v2/src/useSelect.ts [hunk 1]
```diff
@@ -804,10 +827,14 @@ const useSelect = (props: ISelectV2Props, emit) => {
     initStates()
   })
   useResizeObserver(selectRef, handleResize)
-  useResizeObserver(selectionRef, resetSelectionWidth)
+  useResizeObserver(selectionRef, () => {
+    resetSelectionWidth()
+    updateTooltip()
+  })
```
Comment (kooriookami) [inline packages/components/select-v2/src/useSelect.ts:833]: Do you think observe `wrapperRef` is better?

Comment (Fuphoenixes) [reply packages/components/select-v2/src/useSelect.ts]: yeah, it's better

**Review by kooriookami — APPROVED** [review]

**Pull request merged**